packages/node/src/node.ts

```diff
@@ -101,58 +111,56 @@ export class Node {
             "Reached OP_SIGN middleware without generated commitment."
           );
         }
-        context.signature = {
-          v: -1,
-          r: HashZero,
-          s: HashZero
-        };
-        // TODO: Replace the above hack with the below
-        // context.signature = this.signer.signDigest(
-        //   context.commitment.hashToSign()
-        // );
+
+        context.signature = this.signer.signDigest(
+          context.commitment.hashToSign()
+        );
+
         next();
       }
     );
-  }
 
-  private registerIoMiddleware() {
-    this.instructionExecutor.register(
+    instructionExecutor.register(
       Opcode.IO_SEND,
       async (message: ProtocolMessage, next: Function, context: Context) => {
-        await this.messagingService.send(context.outbox[0].toAddress, {
-          from: this.address,
-          event: NODE_EVENTS.PROTOCOL_MESSAGE_EVENT,
-          data: context.outbox[0]
-        });
+        const from = getAddress(this.address);
+        const to = getAddress(context.outbox[0].toAddress);
+        const data = context.outbox[0];
+
+        if (context.outbox[0].seq === 1) {
+          this.ioSendDeferrals[to] = new Deferred<
```

Comment:
there's one place in the `installVirtualApp` protocol that we do two `SEND`s in a row and then immediately end (without any `WAIT` after)

Reply:
I'll implement `IO_SEND_AND_WAIT` in a separate PR.